### PR TITLE
Vault 1.13 updates

### PR DESCRIPTION
**Pull request by tvoran**
go.mod 1.12 => 1.19
hashicorp/vault/api v1.0.5-0.20200215224050-f6547fa8e820 => v1.8.3
hashicorp/vault/sdk v0.1.14-0.20200215224050-f6547fa8e820 => v0.7.0
golang.org/x/crypto v0.0.0-20210921155107-089bfa567519 => v0.5.0
golang.org/x/net 0.0.0-20220722155237-a158d28d115b => v0.5.0

Fix `TestPathKeysRegister_Write()` to use **logical.UpdateOperation** instead of **logical.CreateOperation**.

```console
❯ make test      
ok  	github.com/hashicorp/vault-plugin-secrets-gcpkms	152.141s
?   	github.com/hashicorp/vault-plugin-secrets-gcpkms/cmd/vault-plugin-secrets-gcpkms	[no test files]
?   	github.com/hashicorp/vault-plugin-secrets-gcpkms/test/cleanup	[no test files]
?   	github.com/hashicorp/vault-plugin-secrets-gcpkms/version	[no test files]
?   	github.com/hashicorp/vault-plugin-secrets-gcpkms/version/cmd	[no test files]
```

### CODE REVIEW

File: path_keys_register_test.go [hunk 1]
```diff
@@ -57,13 +56,12 @@ func TestPathKeysRegister_Write(t *testing.T) {
...
 
 			t.Run(tc.name, func(t *testing.T) {
-
 				key := path.Base(tc.cryptoKey)
 
 				b, storage := testBackend(t)
 				_, err := b.HandleRequest(context.Background(), &logical.Request{
 					Storage:   storage,
-					Operation: logical.CreateOperation,
+					Operation: logical.UpdateOperation,
```
Comment (sethvargo) [inline path_keys_register_test.go:64]: Why is this change required?

Comment (benashz) [reply path_keys_register_test.go:64]: > Why is this change required?

I believe that is related to: https://github.com/hashicorp/vault/pull/18492

Comment (benashz) [reply path_keys_register_test.go:64]: I think the failure may have been introduced in #20

**Review by benashz — APPROVED** [review]
LGTM!

**Review by sethvargo — APPROVED** [review]

File: path_keys_register_test.go (hunk 1 above)
Comment (tvoran) [reply path_keys_register_test.go:64]: @benashz Yep, exactly. Without this change the `TestPathKeysRegister_Write` tests were failing with 

```console
path_keys_register_test.go:78: unsupported operation
```

**Comment by tvoran** [comment]
Thanks!
